tests/test_protocol.py

```diff
@@ -1590,3 +1592,88 @@ async def test_methodY(self, id: str, name: str, age: int) -> str:  # NOQA
 
     request = MethodProperties.methods["test_method"][0].build_call(args=[], kwargs={"id": "1", "name": "monty", "age": 42})
     assert request.url == "/api/v1/test/1/monty?age=42"
+
+
+@pytest.mark.asyncio(timeout=5)
+async def test_2151_method_header_parameter_in_body(async_finalizer) -> None:
```

Comment:
I've moved this to `tests/test_protocol.py` because that seems to be the correct place. I had added this to `test_server.py` a couple of weeks ago.